src/main/java/uk/gov/hmcts/reform/idam/health/vault/msi/CustomAppServiceMSICredentials.java

```diff
@@ -0,0 +1,76 @@
+package uk.gov.hmcts.reform.idam.health.vault.msi;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.api.client.http.GenericUrl;
+import com.google.api.client.http.HttpBackOffUnsuccessfulResponseHandler;
+import com.google.api.client.http.HttpHeaders;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestFactory;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.util.ExponentialBackOff;
+import com.microsoft.azure.AzureEnvironment;
+import com.microsoft.azure.credentials.AzureTokenCredentials;
+
+import java.io.IOException;
+
+public class CustomAppServiceMSICredentials extends AzureTokenCredentials {
+
+    private static final String MSI_TOKEN_URL = "http://169.254.169.254/metadata/identity/oauth2/token";
```

Comment:
I think I would pass these values in to the constructor and get them from the properties file (in the same way we used to the get the client id and key)

Reply:
The value is same for every environment.